### ISSUE TITLE
Add 'with Book()' context manager

### ISSUE THREAD

**Issue by fzumstein** [xlwings/xlwings#512]

**Comment by Eupraxis** [comment]
What are the __enter__() and __exit__() needs? Should it try to open a  new workbook (or one with the given name e.g., `with Book("Testbook.xlsx") as wb:`  on entering then save and close upon exit?

**Comment by fzumstein** [comment]
It isn't very clear yet...one thing to consider is that when you do `with Book("Testbook.xlsx") as wb:`, `Testbook.xlsx` can be both, open or closed.

**Comment by Eupraxis** [comment]
Ok. I was just wondering if there was a particular use case that prompted your question. Have you run into instances where a context manager would have been helpful?

**Comment by fzumstein** [comment]
It was a user question so I logged it here. I guess it's somewhat natural to expect a context manager if you're used to using it with text files, let's say.

**Comment by ckp95** [comment]
This is what I've been using:

```
from contextlib import contextmanager
import xlwings as xw@contextmanager
def safe_app(*args, **kwargs):
    app = xw.App(*args, **kwargs)
    try:
        yield app
    finally:
        app.kill()@contextmanager
def freeze(app, screen_updating=False, display_alerts=False, calculation="manual"):
    before = app.screen_updating, app.display_alerts, app.calculation
    try:
        app.screen_updating = screen_updating
        app.display_alerts = display_alerts
        app.calculation = calculation
        yield app
    finally:
        app.screen_updating, app.display_alerts, app.calculation = before@contextmanager
def safe_workbook(app, *args, **kwargs):
    wb = app.books.open(*args, **kwargs)
    try:
        yield wb
    finally:
        wb.close()
```

I use these to reduce boilerplate and to make sure that there aren't leftover Excel processes hanging around if an exception is encountered. Needing two layers of indentation before I can do any work is a little awkward (and more if I need multiple workbooks open at once), but it works. Feel free to use them.

**Comment by fzumstein** [comment]
@ckp95 note that 0.20.5 fixed the issue with the zombie process -- it's now properly cleaned up when calling `app.quit()`.

**Comment by fzumstein** [comment]
Closing this as App() (not Book()) is now a context manager, see #1657, will be released with 0.24.3.
